analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DeliveringDebugFeaturesInProductionBase.cs

```diff
@@ -55,10 +55,7 @@ protected override void Initialize(TrackerInput input)
                     t.ExceptWhen(c => IsInDevelopmentContext(c.Node)));
         }
 
-        protected bool IsValidationMethod(SemanticModel semanticModel, SyntaxNode condition, string methodName)
-        {
-            var methodSymbol = new Lazy<IMethodSymbol>(() => semanticModel.GetSymbolInfo(condition).Symbol as IMethodSymbol);
-            return isDevelopmentMethods.Any(x => x.IsMatch(methodName, methodSymbol, Language.NameComparison));
-        }
+        protected bool IsValidationMethod(SemanticModel semanticModel, SyntaxNode condition, string methodName) =>
+            isDevelopmentMethods.Any(x => x.IsMatch(methodName, new Lazy<IMethodSymbol>(() => semanticModel.GetSymbolInfo(condition).Symbol as IMethodSymbol), Language.NameComparison));
```

Comment:
IMO the inline object initialization with the lambda inside makes it a bit less readable.